packages/slider/src/Slider.ts

```diff
@@ -16,44 +16,69 @@ import {
     CSSResultArray,
     TemplateResult,
     query,
-    PropertyValues,
     styleMap,
     ifDefined,
 } from '@spectrum-web-components/base';
-import { streamingListener } from '@spectrum-web-components/base/src/streaming-listener.js';
 
 import sliderStyles from './slider.css.js';
-import { Focusable } from '@spectrum-web-components/shared/src/focusable.js';
 import { StyleInfo } from 'lit-html/directives/style-map';
+import { HandleController } from './HandleController';
+import { SliderHandleBase, HandleValues } from './SliderHandle';
 
 export const variants = ['filled', 'ramp', 'range', 'tick'];
 
-export class Slider extends Focusable {
+const ValueConverter = {
+    fromAttribute: (value: string): number | HandleValues => {
+        const numeric = parseFloat(value);
+        if (!isNaN(numeric)) return numeric;
+        return JSON.parse(value) as HandleValues;
+    },
+    toAttribute: (value: number | HandleValues) => {
+        if (typeof value === 'number') return value.toString();
+        return JSON.stringify(value);
+    },
+};
+export class Slider extends SliderHandleBase {
     public static get styles(): CSSResultArray {
         return [sliderStyles];
     }
 
-    @property()
-    public type = '';
+    private handleController!: HandleController;
 
-    @property({ type: Number, reflect: true })
-    public get value(): number {
-        return this._value;
+    constructor() {
+        super();
+        this.handleController = new HandleController(this);
     }
 
-    public set value(value: number) {
-        const oldValue = this.value;
-        if (this.input) {
-            this.input.value = String(value);
-        }
-        const newValue = this.input ? parseFloat(this.input.value) : value;
+    @property()
+    public type = '';
 
-        if (newValue === oldValue) {
-            return;
+    @property({ reflect: true, converter: ValueConverter })
```

Comment:
I've had trouble with `@property` working as expected (usually as documented, but you know expectations...) with getter/setter usage. If this works, great, but just something to keep an eye on it.

Reply:
Good to know. I think I got the idea from the original slider code. I will watch it like a hawk 😄

Reply:
It was mainly that either reflection or render didn't happen without actually telling it to. Might be good to focus a test on those two side effects here.

Reply:
I changed to make `value` a simple number attribute